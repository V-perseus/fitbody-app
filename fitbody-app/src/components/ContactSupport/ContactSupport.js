import React from 'react'
import { View, Text } from 'react-native'
import { TouchableOpacity } from 'react-native-gesture-handler'
import globals from '../../config/globals'

const ContactSupport = (props) => {
  return (
    <View style={[styles.container, props.style]}>
      <Text style={[styles.questions, props.color ? { color: props.color } : null]}>Have Questions? </Text>
      <TouchableOpacity onPress={props.onPress}>
        <Text style={[styles.contactus, { color: props.linkColor ? props.linkColor : props.color ? props.color : undefined }]}>
          Contact Us.
        </Text>
      </TouchableOpacity>
    </View>
  )
}

const styles = {
  container: {
    alignSelf: 'stretch',
    justifyContent: 'center',
    flexDirection: 'row',
  },
  questions: {
    fontFamily: globals.fonts.primary.style.fontFamily,
    fontSize: 12,
    letterSpacing: 0.4,
    color: globals.styles.colors.colorWhite,
  },
  contactus: {
    fontSize: 12,
    letterSpacing: 0.4,
    color: globals.styles.colors.colorWhite,
    fontFamily: globals.fonts.primary.boldStyle.fontFamily,
  },
}

export default ContactSupport
