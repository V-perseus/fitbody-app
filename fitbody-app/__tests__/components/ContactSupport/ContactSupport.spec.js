import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import ContactSupport from '../../../src/components/ContactSupport/ContactSupport'

describe('ContactSupport', () => {
  const onPress = jest.fn()

  it('renders snapshot correctly', () => {
    const tree = render(<ContactSupport onPress={onPress} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render', () => {
    const { queryByText } = render(<ContactSupport onPress={onPress} />)
    expect(queryByText(/Have Questions?/)).not.toBeNull()
  })

  it('should handle press event', async () => {
    const { getByText } = render(<ContactSupport onPress={onPress} />)
    await fireEvent.press(getByText('Contact Us.'))
    expect(onPress).toHaveBeenCalled()
  })
})
