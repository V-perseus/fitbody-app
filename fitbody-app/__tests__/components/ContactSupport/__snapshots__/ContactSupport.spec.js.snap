// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSupport renders snapshot correctly 1`] = `
<View
  style={
    [
      {
        "alignSelf": "stretch",
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
          "fontFamily": "OpenSans",
          "fontSize": 12,
          "letterSpacing": 0.4,
        },
        null,
      ]
    }
  >
    Have Questions? 
  </Text>
  <RNGestureHandlerButton
    collapsable={false}
    exclusive={true}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    touchSoundDisabled={false}
  >
    <View
      accessible={true}
      collapsable={false}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 12,
              "letterSpacing": 0.4,
            },
            {
              "color": undefined,
            },
          ]
        }
      >
        Contact Us.
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;
